core/src/main/scala/com/databricks/labs/remorph/parsers/tsql/TSqlExpressionBuilder.scala

```diff
@@ -66,25 +78,83 @@ class TSqlExpressionBuilder
+    }
+  }
+
+  override def visitPrimitiveConstant(ctx: TSqlParser.PrimitiveConstantContext): ir.Expression = {
+    if (ctx.DOLLAR != null) {
+      return ir.Literal(string = Some(ctx.getText))
+    }
+    buildConstant(ctx.con)
+  }
+
+  override def visitScNot(ctx: TSqlParser.ScNotContext): ir.Expression =
+    ir.Not(ctx.searchCondition().accept(this))
+
+  override def visitScAnd(ctx: TSqlParser.ScAndContext): ir.Expression =
+    ir.And(ctx.searchCondition(0).accept(this), ctx.searchCondition(1).accept(this))
+
+  override def visitScOr(ctx: TSqlParser.ScOrContext): ir.Expression =
+    ir.Or(ctx.searchCondition(0).accept(this), ctx.searchCondition(1).accept(this))
+
+  override def visitScPred(ctx: TSqlParser.ScPredContext): ir.Expression = ctx.predicate().accept(this)
+
+  override def visitScPrec(ctx: TSqlParser.ScPrecContext): ir.Expression = ctx.searchCondition.accept(this)
+
+  override def visitPredicate(ctx: TSqlParser.PredicateContext): ir.Expression = {
+    val left = ctx.expression(0).accept(this)
+    val right = ctx.expression(1).accept(this)
+
+    ctx.comparisonOperator match {
+      case op if op.LT != null && op.EQ != null => ir.LesserThanOrEqual(left, right)
+      case op if op.GT != null && op.EQ != null => ir.GreaterThanOrEqual(left, right)
+      case op if op.LT != null && op.GT != null => ir.NotEquals(left, right)
+      case op if op.EQ != null => ir.Equals(left, right)
+      case op if op.GT != null => ir.GreaterThan(left, right)
+      case op if op.LT != null => ir.LesserThan(left, right)
+    }
   }
 
-  override def visitTerminal(node: TerminalNode): ir.Expression = node.getSymbol.getType match {
-    case c if c == STRING => Literal(string = Some(removeQuotes(node.getText)))
-    case c if c == INT => Literal(integer = Some(node.getText.toInt))
-    case c if c == FLOAT => Literal(float = Some(node.getText.toFloat))
-    case c if c == HEX => Literal(string = Some(node.getText)) // Preserve format for now
-    case c if c == REAL => Literal(double = Some(node.getText.toDouble))
-    case c if c == NULL_ => Literal(nullType = Some(ir.NullType()))
-    case _ => wrapUnresolvedInput(node.getText)
+  /**
+   * For now, we assume that we are dealing with Column names. Later we can add some context by keeping a symbol table
+   * for DECLARE. LOCAL_ID is not catered for as part of an expression in the current grammar, but even that can be an
+   * alias for a column name.
+   *
+   * For now then, they are all seen as columns.
+   *
+   * @param ctx
+   *   the parse tree
+   */
+  override def visitId(ctx: IdContext): ir.Expression = ctx match {
+    case c if c.ID() != null => ir.Column(ctx.getText)
+    case c if c.TEMP_ID() != null => ir.Column(ctx.getText)
+    case c if c.DOUBLE_QUOTE_ID() != null => ir.Column(ctx.getText)
+    case c if c.SQUARE_BRACKET_ID() != null => ir.Column(ctx.getText)
+    case c if c.RAW() != null => ir.Column(ctx.getText)
+    case _ => ir.Column(ctx.getText)
```

Comment:
No - unless you want to case every keyword in TSQL. The parser rule Id does not yield anything that cannot be an identifier. If it was not for the quoting, we could just ignore the token type altogether.  So that RAW is only there because it is specifically mentioned. Every other id token, we just need the raw text.